src/v/security/jwt.h

```diff
@@ -351,6 +351,15 @@ class jwt {
     // https://www.rfc-editor.org/rfc/rfc7519#section-4.1.7
     auto jti() const { return claim("jti"); }
 
+    // Return all claim names in the payload
+    chunked_vector<ss::sstring> get_claim_names() const {
+        return {
+          std::from_range,
+          std::views::transform(_payload.GetObject(), [](const auto& m) {
+              return ss::sstring(detail::as_string_view(m.name));
+          })};
```

Comment:
nit: Since this is only used synchronously for logging, a range of `string_view` would avoid some allocations:
```suggestion
    auto get_claim_names() const {
        return std::views::transform(_payload.GetObject(), [](const auto& m) {
            return detail::as_string_view(m.name);
        });
```